src/matches.rs

```diff
@@ -58,6 +64,25 @@ declare_lint!(pub MATCH_REF_PATS, Warn,
 declare_lint!(pub MATCH_BOOL, Warn,
               "a match on boolean expression; recommends `if..else` block instead");
 
+/// **What it does:** This lint checks for overlapping match arms. It is `Warn` by default.
+///
+/// **Why is this bad?** It is likely to be an error and if not, makes the code less obvious.
+///
+/// **Known problems:** None
+///
+/// **Example:**
+///
+/// ```
+/// let x = 5;
+/// match x {
+///     1 ... 10 => println!("1 ... 10"),
+///     5 ... 15 => println!("5 ... 15"),
+///     _ => (),
+/// }
+/// ```
+declare_lint!(pub MATCH_OVERLAPPING_ARM, Warn,
```

Comment:
Do you mean `declare_lint!{…}`? I did what the existing lints in that file did.
There seem to be no consistency in clippy (48 () vs. 39 {}), so it could be better to make another PR for that.